### DISCUSSION TITLE
Using WinForms with the latest CSLA

### DISCUSSION THREAD

**Discussion by aprokopyev** [MarimerLLC/csla#2045] [Questions] [answered]
Hello,

It seems WinForms is marked as deprecated in your nuget packages like 5.0.0-R19040402.

Please let me know, how much work is it to add/merge WinForms support by myself from your earlier releases like 5.2 to the latest release like 5.4 and newer?

I would like to use the same CSLA object model for two GUI applications at once: light Blazor for external web users, and rich WinForms for internal company's users and admins. 

Is it still possible to easy update your latest releases of CSLA to keep them compatible with WinForms yet?

There is still a lot of work done to actively develop and support very good WinForms components like:
https://community.devexpress.com/tags/WinForms/default.aspx

As for Blazor I would prefer Syncfusion (free for small groups and individual developers) or Telerik components.

**Comment by rockfordlhotka** [comment, marked as answer]
I just checked and that is not the case.

https://www.nuget.org/packages/Csla.Windows.Forms

(I was worried that maybe some current version got marked that way by accident)

I've been marking old versions of CSLA as deprecated, slowly over time because I keep running out of nuget API calls, and there are _a lot_ of old versions out there when you count all the preview releases!

But version 5.4 is there and not marked as deprecated.

**Reply by aprokopyev** [reply]
Thank you very much for the clarification, IDE offered me to add this package:
https://www.nuget.org/packages/CSLA-WindowsForms/5.0.0-R19040402

Now I see a newer package v5.4 which is not deprecated.

May I know for how long are you going to support WinForms in newer CSLA version like v6, ..., etc.? 
According to:
https://blog.submain.com/death-winforms-greatly-exaggerated/
http://www.freezepage.com/1610705012EWJZKBJQOX
there are still a lot of developers using WinForms even in new projects.

**Reply by rockfordlhotka** [reply]
All the old packages with a dash ("-") in them have been deprecated and are only for versions prior to CSLA 5.
